Add main panel

vlnt-trsv opened the pull request.

vlnt-trsv committed:
feat(home): add way menu child component

vlnt-trsv committed:
fix(store): change name & to link

vlnt-trsv committed:
feat(home): add sale ticket block

vlnt-trsv committed:
fix(home): add outlet

vlnt-trsv committed:
fix(home): add active link to nav panel

vlnt-trsv committed:
fix(ui): styling & refactor

vlnt-trsv committed:
fix(hooks): refactor useStepper

vlnt-trsv committed:
feat(assets): add icons

vlnt-trsv committed:
fix(root): add white theme for text

vlnt-trsv committed:
fix(router): add sale ticket route

vlnt-trsv committed:
Add way menu (#33)

**Без обратного билета**


![image](https://github.com/user-attachments/assets/da997e5f-6852-462e-9a9c-ee53d4f522e4)

**С обратным билета**


![image](https://github.com/user-attachments/assets/972a02f1-f073-4fe9-be71-abd545c0423a)


Компонент WayMenu предназначен для поиска маршрутов исходя откуда, куда
и когда параметров.
Выводит список маршрутов

vlnt-trsv committed:
feat(home): add way main item & list

vlnt-trsv committed:
fix(home): refactor

vlnt-trsv committed:
fix(store): add type & new void

vlnt-trsv committed:
fix(root): change root

vlnt-trsv committed:
fix(ui): styling & refactor

vlnt-trsv committed:
feat(utils): add get format date

vlnt-trsv committed:
Add way main item (#34)

![image](https://github.com/user-attachments/assets/27dfbe2e-7bcc-435d-94da-f01c0d1d2dc2)

Компонент WayMainItem предназначен для пользовательского интерфейса.
Компонент WayMainList выводит список WayMainItem для выбора одного из
них

vlnt-trsv committed:
fix(ui): styling & refactor

vlnt-trsv committed:
feat(home): add seat main item

vlnt-trsv committed:
fix(home): refactor

vlnt-trsv committed:
fix(store): add types & optimize

vlnt-trsv committed:
chore(package): add immer for store

vlnt-trsv committed:
Add seat main item (#36)

![image](https://github.com/user-attachments/assets/f089ad51-54d9-4a65-8576-e2451d943539)

Компонент SeatMainItem предназначен для выбора мест на конкретном
маршруте.

vlnt-trsv committed:
fix(root): change root for chip & select

vlnt-trsv committed:
feat(home): add passenger info item

vlnt-trsv committed:
fix(home): styling & optimize

vlnt-trsv committed:
fix(store): adding types & void

vlnt-trsv committed:
fix(assets): add icons

vlnt-trsv committed:
fix(ui): styling & refactor

vlnt-trsv committed:
Add passenger info item (#37)

![image](https://github.com/user-attachments/assets/3730c5a0-605c-4055-bc6e-ff30113d2e23)

Компонент PassengerInfoItem предназначен для ввода данные пассажира на
конкретном маршруте.

Bakugan74 commented on src/pages/home/components/mainPanel/child/saleTicket/SaleTicket.tsx:
```diff
@@ -0,0 +1,98 @@
+      });
+      nextStep();
+    }
+  };
```
может будет избегать else if?

vlnt-trsv replied:
Какие есть предложения?

vlnt-trsv replied:
Можно через switch case 

```suggestion
 const handleNextStep = () => {
  switch (true) {
    case activeStep === 0 && !activeWay:
      alert("Выберите маршрут");
      return;
    case activeStep === 1 && activeWay?.seatsSelected.length === 0:
      alert("Выберите места");
      return;
    case activeStep === 2 && passengers.some((p) => !p.ticket):
      alert("Заполните данные пассажира");
      return;
    default:
      setSearchParams({
        step: String(activeStep + 1),
      });
      nextStep();
  }
};
```

vlnt-trsv committed:
chore(package): add react-query

vlnt-trsv committed:
feat(types): add types for store

vlnt-trsv committed:
feat(app): add query client

vlnt-trsv committed:
fix(root): styling

vlnt-trsv committed:
fix(home): refactor & optimize

vlnt-trsv committed:
feat(home): add way payment

vlnt-trsv committed:
fix(store): remove types & refactor

vlnt-trsv committed:
fix(ui): styling

vlnt-trsv committed:
Add way payment (#39)


![image](https://github.com/user-attachments/assets/c238f1a6-dace-4125-9155-b1acdf99c731)

Компонент WayPayment предназначен для оплаты конкретного маршрута
